aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -95,20 +95,21 @@ void TensorIterator::compute_types() {
   if (missing_dtypes || compute_common_dtype_) {
     auto& type = compute_common_type();
     for (auto& op : operands_) {
+      auto& op_tensor_type = at::globalContext().getNonVariableType(op.tensor.type().backend(), op.tensor.type().scalarType());
       if (!op.type) {
         op.type = &type;
```

Comment:
@gchanan the symptom is that `type` here will always be a non-Variable type because of https://github.com/pytorch/pytorch/pull/16325/files#diff-b47a50873394e38a005b4c1acd151957L155, but at https://github.com/pytorch/pytorch/pull/16325/files#diff-b47a50873394e38a005b4c1acd151957L120 we are comparing `op.tensor.type()` to `op.type`, and `op.tensor.type()` is not a non-Variable type unless we explicitly get the non-Variable type from it (which is what the changes in this file aim to do)

Reply:
I see, it's probably cleaner to just store the backend + dtype, but you don't need to make that change.